Linked list with void pointers.c

```diff
@@ -5,29 +5,26 @@ typedef struct node
 {
     void *data;
     struct node *next;
+    void (*datatype)();
 } node;
-struct node *head = NULL;
-struct node *tail = NULL;
-void insert(node **head, node **tail, const void * pointer, size_t size)
+typedef struct  LLhead
 {
-    if (*head == NULL)
-    {
-        *head = malloc(sizeof(node));
-        (*head)->data = malloc(size);
-        memcpy((*head)->data, pointer, size);
-        (*head)->next = NULL;
-        *tail = *head;
-    }
-    else
-    {
-        node *temp;
-        temp = malloc(sizeof(node));
-        temp->data = malloc(size);
-        memcpy(temp->data, pointer, size);
-        temp->next = *tail;
-        (*tail) = temp;
- 
-    }
+    struct node *head ;
+    struct node *tail ;
+} LLhead;
+
+void insert(node **tail,void(*display_datatype)(node *tail), void * pointer,int (*function)(void *ptr))

```

Comment:
Insert should take LLhead itself, 
what is pointer ? 
Read the whole code again and make sure every variable name is good.
What is the difference between pointer and *ptr? Does anyone understand it ? Correct them